### ISSUE TITLE
Can't deploy from "src" to "master"?

### ISSUE THREAD

**Issue by kud** [shinnn/gulp-gh-pages#54]
Hello,

I've got the same script for two blogs. One is on a classic repo and the other one is the username.github.io.

The first one works like a charm, the second is broken. See

```console
[01:35:26] Error in plugin 'gulp-gh-pages'
Message:
    Command failed: /bin/sh -c git rev-list --pretty=raw --skip=0 --max-count=10 src
fatal: ambiguous argument 'src': both revision and filename
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

Details:
    killed: false
    code: 128
    signal: null
    cmd: /bin/sh -c git rev-list --pretty=raw --skip=0 --max-count=10 src
```

Script:

```javascript
module.exports = function() {

  var gulp   = require('gulp')
  var deploy = require('gulp-gh-pages')

  return gulp.src('dist/**/*')
    .pipe(
      deploy({
        branch: 'master',
        cacheDir: '.deploy'
      })
    )

}
```

Note that what is copied in '.deploy' isn't the result of 'dist/` but what I've got in the root of my project (except `dist`).

Do you have any idea why? -> https://github.com/kud/kud.github.io

**Comment by shinnn** [comment]
I found the cause http://stackoverflow.com/a/26349250

We should open a new issue or create PR for [gift](https://github.com/notatestuser/gift) to fix this problem.

**Comment by kud** [comment]
Wow, it's caused by the fact i've got `src` folder and `src` branch. I don't know why but I felt it was something like that. Probably because I spend lots of time on computers.

Anyway, the workaround is to call my branch or my folder in another name.

Thank you @shinnn for this!

**Comment by kud** [comment]
I've changed my branche from `src` to `sources` and it indeed works. Be sure to remove `.publish` too.

**Comment by shinnn** [comment]
@kud I'm glad I was able to help you.

I want to fix the root cause but [gift](https://github.com/notatestuser/gift) project seems to be abandoned.

**Comment by jboothe** [comment]
@kud Are you saying to remove .publish before committing any branch?

**Comment by kud** [comment]
No. I'm saying that if you have some troubles, do not forget to remove `.publish` because it can fix it. ;)

**Comment by shinnn** [comment]
Fixed in [v0.6.0-5](https://github.com/shinnn/gulp-gh-pages/tree/v0.6.0-5). You'll never see this error as this plugin no longer executes `git rev-list`.

Thanks for the report, @kud!
